src/connectors/tests/elastic.rs

```diff
@@ -155,6 +155,71 @@ async fn connector_elastic() -> Result<()> {
         event.data.suffix().value()
     );
 
+    // upsert
+
+    let data = literal!({
+        "doc": {
+            "field1": 13.4,
+            "field2": "strong",
+            "field3": [false, true],
+            "field4": {
+                "nested": false
+            }
+        },
+        "doc_as_upsert": true,
+    });
+    let meta = literal!({
+        "elastic": {
+            "_index": "my_index",
+            "_id": "1234",
+            "action": "update",
+            "raw_payload": true
+        },
+    });
+    let event_not_batched = Event {
+        id: EventId::default(),
+        data: (data, meta).into(),
+        transactional: false,
+        ..Event::default()
+    };
+    harness.send_to_sink(event_not_batched, IN).await?;
+    let err_events = err.get_events()?;
+    assert!(err_events.is_empty(), "Received err msgs: {:?}", err_events);
+    let event = out.get_event().await?;
+    assert_eq!(
+        &literal!({
+            "elastic": {
+                "_id": "1234",
+                "_index": "my_index",
+                "_type": "_doc",
+                "version": 1,
+                "action": "update",
+                "success": true
+            },
+        }),
+        event.data.suffix().meta()
+    );
+    assert_eq!(
+        &literal!({
+            "update": {
+                "_primary_term": 1,
+                "_shards": {
+                    "total": 2,
+                    "successful": 1,
+                    "failed": 0,
+                },
+                "_type": "_doc",
+                "_index": "my_index",
+                "result": "created",
+                "_id": "1234",
+                "_seq_no": 1,
+                "status": 201,
+                "_version": 1
+            }
+        }),
+        event.data.suffix().value()
+    );
+
```

Comment:
It would be cool to actually verify that the `doc_as_upsert` works, by upserting another document with different fields and check that the resulting document is a mix/merge of the inserted and updated doc.

Reply:
It's verifying it by updating a document that doesn't exist, which update would fail, and succeeding. We're not validating content for `insert` either I don't think adding this requirement here makes sense.

Reply:
Alrighty!